certora/specs/ExchangeRate.spec

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (c) 2025 Morpho Association
+
+import "Invariants.spec";
+
+definition shares() returns mathint = currentContract.totalSupply + currentContract.virtualShares;
+
+definition assets() returns mathint = currentContract._totalAssets + 1;
+
+// Check that if deposit adds one more share to the user than it does, then the share price would decrease following a deposit.
+rule sharePriceBoundDeposit(env e, uint256 assets, address onBehalf){
+    require (e.block.timestamp == currentContract.lastUpdate, "assume no interest is accrued");
+
+    mathint assetsBefore = assets();
+    mathint sharesBefore = shares();
+
+    deposit(e, assets, onBehalf);
+
+    assert assets() * sharesBefore <= assetsBefore * (shares() + 1);
```

Comment:
I think this should also hold with `<` instead of `<=` and it would match the rule comment (and similarly for the other "Bound" rules).

Although, the rule comment itself is kind of confusing.  It says, if the protocol would give the user one share more, it would lose money (share price decreases). The real property the user is interested in is that the user gets at most one share less than the fair amount.  Both properties are equivalent to each other, though.  And since we need to avoid rational numbers, you can only express them indirectly as an integer disequality after multiplying with all the denominators.